
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Command line tools &mdash; SPiCE 0.1 documentation</title>
    
    <link rel="stylesheet" href="_static/spicedoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="SPiCE 0.1 documentation" href="index.html" />
    <link rel="prev" title="Example projects" href="example_projects.html" />
    <link href="http://fonts.googleapis.com/css?family=Ubuntu+Mono|Ubuntu|Raleway" rel="stylesheet" type="text/css">
     

  </head>
  <body>
    <!--div id="header_menu">
        <a href=http://helix.ewi.tudelft.nl/spica/news/>news</a>
        <a href=http://helix.ewi.tudelft.nl/spica/doc/ class="active">documentation</a>
        <a href=http://helix.ewi.tudelft.nl/spica/software/>software</a>
        <a href=http://helix.ewi.tudelft.nl/spica/about/>about</a>
        <a href=http://helix.ewi.tudelft.nl/spica/contact/>contact</a>
    </div-->
    <div id="header" class="home">
        <div id="title"><!--a href="http://helix.ewi.tudelft.nl/spice"-->SPiCe<!--/a--></div>
        <div id="subtitle">Sequence-based Protein Classification and
Exploration</div>
        <div id="version">version 0.1</div>
    </div>
    

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="example_projects.html" title="Example projects"
             accesskey="P">previous</a> |</li>
    
        <li><a href="index.html">SPiCE 0.1 documentation</a> &raquo;</li>
 
      </ul>
    </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Command line tools</a><ul>
<li><a class="reference internal" href="#featext-py">featext.py</a></li>
<li><a class="reference internal" href="#classification-py">classification.py</a></li>
<li><a class="reference internal" href="#classify-py">classify.py</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="example_projects.html"
                        title="previous chapter">Example projects</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/command_line_tools.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="command-line-tools">
<span id="id1"></span><h1>Command line tools<a class="headerlink" href="#command-line-tools" title="Permalink to this headline">¶</a></h1>
<p>SPiCE can also be used locally using the command line using Python package
<a class="reference external" href="https://github.com/basvandenberg/spice">spice</a>. This package contains a feature extraction (<tt class="docutils literal"><span class="pre">featext</span></tt>) module that
can be used for features calculation, a <tt class="docutils literal"><span class="pre">classification</span></tt> module to train
classifiers and a <tt class="docutils literal"><span class="pre">classify</span></tt> module to run a trained classifier on another
data set.</p>
<div class="section" id="featext-py">
<h2>featext.py<a class="headerlink" href="#featext-py" title="Permalink to this headline">¶</a></h2>
<p>For feature extraction, a project needs to be initialized first at a give
location (path). For example:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">mkdir</span> <span class="n">test_project</span>
<span class="n">cd</span> <span class="n">test_project</span>
<span class="n">featext</span> <span class="o">--</span><span class="n">root</span> <span class="o">.</span> <span class="o">--</span><span class="n">init</span>
</pre></div>
</div>
<p>A directory protein_data_set has been created, which will contain all protein
sequence data. The directory feature_matrix_protein will contain all calculated
features in the form of a feature matrix.</p>
<p>Next the protein ids need to be added. These should be stored in a file that
contains exactly one (unique) protein id per line.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">featext</span> <span class="o">--</span><span class="n">root</span> <span class="o">.</span> <span class="o">--</span><span class="n">uniprot_ids</span> <span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">my</span><span class="o">/</span><span class="n">ids_file</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>The option is called uniprot_ids for historic reasons, but you can use any ids
you like. This naming will be adjusted in a following release.</p>
<p>A protein_ids.txt file is now stored in the protein_data_set directory. The
next step is to add protein sequences, using a FASTA-file. This file should at
least contain a sequence for each id you just added to the project. Only these
sequences will be read and added to the projects protein data.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">featext</span> <span class="o">--</span><span class="n">root</span> <span class="o">.</span> <span class="o">--</span><span class="n">protein_seqeunce_data</span> <span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">my</span><span class="o">/</span><span class="n">fasta_file</span><span class="o">.</span><span class="n">fsa</span>
</pre></div>
</div>
<p>A protein.fsa file is now stored in the protein_data_set directory. This
sequence data can now be used for calculating features.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">featext</span> <span class="o">--</span><span class="n">root</span> <span class="o">.</span> <span class="o">--</span><span class="n">protein_features</span> <span class="n">aac_1</span> <span class="n">dc_1</span>
</pre></div>
</div>
<p>The protein_features options <tt class="docutils literal"><span class="pre">aac_1</span></tt> and <tt class="docutils literal"><span class="pre">dc_1</span></tt> are feature category ids
that indicate what features need to be calculated. More information about
feature ids can be found in the <a class="reference internal" href="features.html#feature-ids"><em>Feature ids</em></a> section.</p>
<p>Protein labels can be set using a <em class="xref std std-ref">labeling file</em>, which is required for
running classification jobs.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">featext</span> <span class="o">--</span><span class="n">root</span> <span class="o">.</span> <span class="o">--</span><span class="n">labels</span> <span class="n">protein</span> <span class="n">my_labeling</span> <span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">my</span><span class="o">/</span><span class="n">labeling_file</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>There are three parameters that should follow the <tt class="docutils literal"><span class="pre">--labels</span></tt> option. The
first should always be protein, the second is the name of your labeling (you
can choose any), and the third is the path to your labeling file.</p>
</div>
<div class="section" id="classification-py">
<h2>classification.py<a class="headerlink" href="#classification-py" title="Permalink to this headline">¶</a></h2>
<p>Classification jobs can be run using:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">classification</span> <span class="o">-</span><span class="n">f</span> <span class="n">your_project_dir</span><span class="o">/</span><span class="n">feature_matrix_protein</span> \
    <span class="o">-</span><span class="n">l</span> <span class="n">labeling_name</span> <span class="o">-</span><span class="n">c</span> <span class="n">lda</span> <span class="o">-</span><span class="n">n</span> <span class="mi">10</span> <span class="o">-</span><span class="n">e</span> <span class="n">roc_auc</span> <span class="o">--</span><span class="n">classes</span> <span class="n">low</span> <span class="n">high</span> \
    <span class="o">--</span><span class="n">features</span> <span class="n">aac_1_A</span> <span class="n">aac_1_C</span> <span class="n">aac_1_D</span> <span class="o">--</span><span class="n">standardize</span> <span class="o">-</span><span class="n">o</span> <span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">output_dir</span>
</pre></div>
</div>
<p>The <tt class="docutils literal"><span class="pre">-f</span></tt> should point to the feature matrix directory within your project
dir. With the <tt class="docutils literal"><span class="pre">-l</span></tt> and <tt class="docutils literal"><span class="pre">--classes</span></tt> options you can indicate which labaling
to use, and within this labeling, which labels should be classified. The <tt class="docutils literal"><span class="pre">-c</span></tt>
option indicates what classifier to use, the <tt class="docutils literal"><span class="pre">-n</span></tt> option holds the number of
CV-loops to use for performance assessment, the <tt class="docutils literal"><span class="pre">-e</span></tt> option holds the
evaluation score to use. The features to use for classification can be
indicating using the <tt class="docutils literal"><span class="pre">--features</span></tt> option. The <tt class="docutils literal"><span class="pre">-standardize</span></tt> option will
most often be used, this causes the feature matrix to be standardized. Finally,
the <tt class="docutils literal"><span class="pre">-o</span></tt> option points to the directory to which the results should be
written.</p>
</div>
<div class="section" id="classify-py">
<h2>classify.py<a class="headerlink" href="#classify-py" title="Permalink to this headline">¶</a></h2>
<p>Downloaded classification results contain, besides performance scores and used
settings, a Scikit-learn classifier that was trained on the entire data set:
<tt class="docutils literal"><span class="pre">classifier.joblib.pkl</span></tt>. This file is pickled using the <tt class="docutils literal"><span class="pre">joblib</span></tt> module
which is offered as part of the <tt class="docutils literal"><span class="pre">scikit-learn</span></tt> package.</p>
<p>Loading and running this classifier could be done with a python script like the
following:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">sklearn.externals</span> <span class="kn">import</span> <span class="n">joblib</span>
<span class="n">clf</span> <span class="o">=</span> <span class="n">joblib</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="s">&#39;path/to/classifier.joblib.pkl&#39;</span><span class="p">)</span>

<span class="sd">&#39;&#39;&#39;</span>
<span class="sd">data = use featext to calculate features for your proteins, this should</span>
<span class="sd">       be the same features as those used for training the classifier.</span>
<span class="sd">&#39;&#39;&#39;</span>

<span class="c"># prediction class labels on data set</span>
<span class="n">pred</span> <span class="o">=</span> <span class="n">clf</span><span class="o">.</span><span class="n">predict</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>

<span class="c"># obtain class probabilities (if possible)</span>
<span class="k">if</span><span class="p">(</span><span class="nb">hasattr</span><span class="p">(</span><span class="n">clf</span><span class="p">,</span> <span class="s">&#39;predict_proba&#39;</span><span class="p">)):</span>
    <span class="n">proba</span> <span class="o">=</span> <span class="n">clf</span><span class="o">.</span><span class="n">predict_proba</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
    <span class="c"># only works for 2-class classification!</span>
    <span class="n">proba</span> <span class="o">=</span> <span class="n">proba</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">]</span>

<span class="c"># obtain decision function output</span>
<span class="k">if</span><span class="p">(</span><span class="nb">hasattr</span><span class="p">(</span><span class="n">clf</span><span class="p">,</span> <span class="s">&#39;decision_function&#39;</span><span class="p">)):</span>
    <span class="n">decision</span> <span class="o">=</span> <span class="n">clf</span><span class="o">.</span><span class="n">decision_function</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="example_projects.html" title="Example projects"
             >previous</a> |</li>
    
        <li><a href="index.html">SPiCE 0.1 documentation</a> &raquo;</li>
 
      </ul>
    </div>
    
    <div class="footer">
        &copy; Copyright 2013, Bastiaan van den Berg.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b1.
    </div>
    <div class="spica_footer">
    Delft University of Technology, <a href="http://bioinformatics.tudelft.nl">Delft Bioinformatics Lab</a>
    </div>

  </body>
</html>