
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>SPiCE features &mdash; SPiCE 0.1 documentation</title>
    
    <link rel="stylesheet" href="_static/spicedoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="SPiCE 0.1 documentation" href="index.html" />
    <link rel="next" title="File Formats" href="file_formats.html" />
    <link rel="prev" title="Feature data sources" href="feature_data_sources.html" />
    <link href="http://fonts.googleapis.com/css?family=Ubuntu+Mono|Ubuntu|Raleway" rel="stylesheet" type="text/css">
     

  </head>
  <body>
    <!--div id="header_menu">
        <a href=http://helix.ewi.tudelft.nl/spica/news/>news</a>
        <a href=http://helix.ewi.tudelft.nl/spica/doc/ class="active">documentation</a>
        <a href=http://helix.ewi.tudelft.nl/spica/software/>software</a>
        <a href=http://helix.ewi.tudelft.nl/spica/about/>about</a>
        <a href=http://helix.ewi.tudelft.nl/spica/contact/>contact</a>
    </div-->
    <div id="header" class="home">
        <div id="title"><!--a href="http://helix.ewi.tudelft.nl/spice"-->SPiCe<!--/a--></div>
        <div id="subtitle">Sequence-based Protein Classification and
Exploration</div>
        <div id="version">version 0.1</div>
    </div>
    

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="file_formats.html" title="File Formats"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="feature_data_sources.html" title="Feature data sources"
             accesskey="P">previous</a> |</li>
    
        <li><a href="index.html">SPiCE 0.1 documentation</a> &raquo;</li>
 
      </ul>
    </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">SPiCE features</a><ul>
<li><a class="reference internal" href="#amino-acid-composition">Amino acid composition</a></li>
<li><a class="reference internal" href="#dipeptide-composition">Dipeptide composition</a></li>
<li><a class="reference internal" href="#terminal-end-amino-acid-count">Terminal end amino acid count</a></li>
<li><a class="reference internal" href="#sequence-length">Sequence length</a></li>
<li><a class="reference internal" href="#property-composition-transition-distribution-ctd">Property Composition / Transition / Distribution (CTD)</a></li>
<li><a class="reference internal" href="#autocorrelation">Autocorrelation</a></li>
<li><a class="reference internal" href="#signal-average">Signal average</a></li>
<li><a class="reference internal" href="#signal-peaks-area">Signal peaks area</a></li>
<li><a class="reference internal" href="#pseudo-amino-acid-composition-type-1">Pseudo amino acid composition type 1</a></li>
<li><a class="reference internal" href="#pseudo-amino-acid-composition-type-2">Pseudo amino acid composition type 2</a></li>
<li><a class="reference internal" href="#quasi-sequence-order-descriptors">Quasi sequence-order descriptors</a></li>
<li><a class="reference internal" href="#secondary-structure-composition">Secondary structure composition</a></li>
<li><a class="reference internal" href="#per-secondary-structure-amino-acid-composition">Per secondary structure amino acid composition</a></li>
<li><a class="reference internal" href="#solvent-accessibility-composition">Solvent accessibility composition</a></li>
<li><a class="reference internal" href="#per-solvent-accessibility-class-amino-acid-composition">Per solvent accessibility class amino acid composition</a></li>
<li><a class="reference internal" href="#codon-composition">Codon composition</a></li>
<li><a class="reference internal" href="#codon-usage">Codon usage</a></li>
<li><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="feature_data_sources.html"
                        title="previous chapter">Feature data sources</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="file_formats.html"
                        title="next chapter">File Formats</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/features_spice.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="spice-features">
<span id="features-spice"></span><h1>SPiCE features<a class="headerlink" href="#spice-features" title="Permalink to this headline">¶</a></h1>
<p>After project initiation, you can use the <em>Calculate</em> button under the
<em>Features</em> tab to calculate a range of sequence-based protein features.</p>
<a class="reference internal image-reference" href="_images/featcalc0.png"><img alt="_images/featcalc0.png" class="align-center" src="_images/featcalc0.png" style="width: 640px;" /></a>
<p>The following sections describe the different feature categories that can be
calculated on the SPiCE website. Some of these features were used in our
previous research <a class="reference internal" href="#vandenberg2010" id="id1">[16]</a> <a class="reference internal" href="#vandenberg2012" id="id2">[17]</a>, and the rest
was extracted from other feature calculation methods <a class="reference internal" href="#li2006" id="id3">[4]</a>
<a class="reference internal" href="#rao2011" id="id4">[13]</a>, <a class="reference internal" href="#shen2008" id="id5">[6]</a>, <a class="reference internal" href="#cao2013" id="id6">[1]</a>.</p>
<div class="section" id="amino-acid-composition">
<h2>Amino acid composition<a class="headerlink" href="#amino-acid-composition" title="Permalink to this headline">¶</a></h2>
<p>The amino acids composition calculates the relative frequency of occurrence for
each of the 20 amino acids in a protein sequence.</p>
<a class="reference internal image-reference" href="_images/featcalc1.png"><img alt="_images/featcalc1.png" class="align-center" src="_images/featcalc1.png" style="width: 640px;" /></a>
<p>Having 20 amino acids, this will result in 20 features. Any other than the 20
unambiguous amino acid letters will be ignored.</p>
<p>To illustrate this, consider the following (unrealistic) protein sequence,
which contains two occurrences of each amino acid in which each amino acid
obtains the value 0.05:</p>
<div class="highlight-python"><pre>&gt;&gt;aac_test
MMAARRNNDDCCEEQQGGHHIILLKKFFPPSSTTWWYYVV</pre>
</div>
<p>will result in the following feature vector:</p>
<div class="highlight-python"><pre>.              A1          R1       ...        V1

aac_test      0.05        0.05      ...       0.05</pre>
</div>
<p>As a parameter, users can specify in how many (equal sized) segments a protein
should be divided, before calculating the amino acid composition of each
segment separately. The number of features will therefore be the number of
segments times 20.</p>
<p>With 2 as number of segments parameter, the sequence will be split in two:</p>
<div class="highlight-python"><pre>segment 1              segment2
MMAARRNNDDCCEEQQGGHH | IILLKKFFPPSSTTWWYYVV</pre>
</div>
<p>The amino acid composition of both segments is calculated which together
results in a feature vector with 40 features:</p>
<div class="highlight-python"><pre>.              A1   ...   V1   ...   A2   ...   V2

aac_test      0.10  ...  0.00  ...  0.00  ...  0.10</pre>
</div>
</div>
<div class="section" id="dipeptide-composition">
<h2>Dipeptide composition<a class="headerlink" href="#dipeptide-composition" title="Permalink to this headline">¶</a></h2>
<p>Similar to the amino acid composition, the dipeptide composition calculates the
relative frequency of occurrence of each of the 400 possible dipeptides in a
protein sequence.</p>
<a class="reference internal image-reference" href="_images/feat_dipep.png"><img alt="_images/feat_dipep.png" class="align-center" src="_images/feat_dipep.png" style="width: 640px;" /></a>
<p>Having 400 possible amino acid pairs (dipeptides), this will result in 400
features. Dipeptides containing any other than the 20 unambiguous amino acid
letters will be ignored.</p>
<p>As a parameter, users can specify in how many (equal sized) segments a protein
should be divided, before calculating the dipeptide composition of each
segment separately. To limit the number of features and to avoid too sparse
feature matrices, the maximal number of segments is set to 2.</p>
<p>For an example sequence that contains 10 dipeptides:</p>
<div class="highlight-python"><pre>&gt;&gt;dc_test
MAAARRNNDDC</pre>
</div>
<p>The resulting feature vector will be:</p>
<div class="highlight-python"><pre>.           AA      AR   ...   AV      RA      RR         VV

dc_test    0.20    0.10  ...  0.00    0.00    0.10  ...  0.00</pre>
</div>
</div>
<div class="section" id="terminal-end-amino-acid-count">
<h2>Terminal end amino acid count<a class="headerlink" href="#terminal-end-amino-acid-count" title="Permalink to this headline">¶</a></h2>
<p>This feature category returns the amino acid counts of a fixed length sequence,
either at the N- or the C-terminal end of the protein sequence.</p>
<a class="reference internal image-reference" href="_images/featcalc2.png"><img alt="_images/featcalc2.png" class="align-center" src="_images/featcalc2.png" style="width: 640px;" /></a>
<p>For example sequence:</p>
<div class="highlight-python"><pre>&gt;&gt;teraac_test
MMAARRNNDDCCEEQQGGHHIILLKKFFPPSSTTWWYYVV</pre>
</div>
<p>the amino acid counts of the length 10 C-terminal end result in the following
feature vector:</p>
<div class="highlight-python"><pre>.            A   R   N   D   C   E   Q   G   H  ...  M  ...  V

teraac_test  2   2   2   2   0   0   0   0   0  ...  2  ...  0</pre>
</div>
<p>in which the counts for M, A, R, N, and D are set to 2, while the remaining 15
features are set to 0.</p>
</div>
<div class="section" id="sequence-length">
<h2>Sequence length<a class="headerlink" href="#sequence-length" title="Permalink to this headline">¶</a></h2>
<p>This category calculates only one feature, the length of the amino acid
sequence.</p>
<a class="reference internal image-reference" href="_images/feat_len.png"><img alt="_images/feat_len.png" class="align-center" src="_images/feat_len.png" style="width: 640px;" /></a>
<p>The example sequence:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="go">&gt;&gt;&gt;len_test</span>
<span class="go">MMAARRNNDD</span>
</pre></div>
</div>
<p>Will result in the following feature vector:</p>
<div class="highlight-python"><pre>.           len

len_test     10</pre>
</div>
</div>
<div class="section" id="property-composition-transition-distribution-ctd">
<h2>Property Composition / Transition / Distribution (CTD)<a class="headerlink" href="#property-composition-transition-distribution-ctd" title="Permalink to this headline">¶</a></h2>
<p>The Composition, Transition, Distribution feature is introduced in
<a class="reference internal" href="#dubchak1995" id="id7">[5]</a>, the implementation is based on the description in
<a class="reference internal" href="#li2006" id="id8">[4]</a>.</p>
<p>For these features, the protein sequence is first translated
from the 20 letter amino acid alphabet to a 3 letter alphabet, in which the 20
amino acids are divided over the three letters based on some property. The
properties and corresponding subdivision of the amino acids are:</p>
<div class="highlight-python"><pre>property                letter A            letter B            letter C
---------------------------------------------------------------------------
hydrophobicity          RKEDQN              GASTPHY             CLVIMFW

normalized v.d. Waal    GACSTPD             NVEQIL              MHKFRYW

polarity                LIFWCMVY            PATGS               HQRKNED

polarizability          GASDT               CPNVEQIL            KMHFRYW

charge                  KR                  ANCQGHILMFPSTWYV    DE

secondary structure     EALMQKRH            VIYCWFT             GNPSD

solvent accessibility   ALFCGIVW            PKQEND              MRSTHY</pre>
</div>
<p>To illustrate this, using charge as property, an amino acid sequence will be
mapped to a three letter charge alphabet as follows:</p>
<div class="highlight-python"><pre>MPMDQSISSPLFPMEKDIDIPLDATPLAQSSSLQLFIHLAEPVVFLQGFDPQKTEYPSVVLRGCLVVRIL
   |          |:| |   |                 |        |  : |      :     :
BBBCBBBBBBBBBBCACBCBBBCBBBBBBBBBBBBBBBBBCBBBBBBBBCBBABCBBBBBBABBBBBABB</pre>
</div>
<p>For the mapped sequence, three types of features are calculated. First the
property letter composition:</p>
<div class="highlight-python"><pre>composition A:  4 / 70 = 0.057

composition B: 58 / 70 = 0.114

composition C:  8 / 70 = 0.829</pre>
</div>
<p>Secondly the relative frequency of occurrence of letter transitions. For
letters A and B this is the number of transitions from A to B and from B to A
divided by the sequence length - 1:</p>
<div class="highlight-python"><pre>transition A-B B-A:  6 / 69 = 0.087

transition A-C C-A:  2 / 69 = 0.029

transition B-C C-B: 14 / 69 = 0.203</pre>
</div>
<p>Finally the property letter distribution is captured by 5 features per
property letter. If we consider letter C, the first feature is the (procentual)
sequence position of the first occurrence of the C:</p>
<div class="highlight-python"><pre>distribution C first:  4 / 70 = 0.057</pre>
</div>
<p>The following 4 features are the (procentual) sequence positions where
respectively 25%, 50%, 75%, and 100% of the letters C is on and before this
position:</p>
<div class="highlight-python"><pre>distribution C  25%: 15/70 = 0.214

distribution C  50%: 19/70 = 0.271

distribution C  75%: 41/70 = 0.586

distribution C 100%: 55/70 = 0.786</pre>
</div>
<p>The same five features are calculated for letters A and B as well. In total the
CTD feature category provides 3 + 3 + 5 x 3 = 21 features for a given property.</p>
</div>
<div class="section" id="autocorrelation">
<h2>Autocorrelation<a class="headerlink" href="#autocorrelation" title="Permalink to this headline">¶</a></h2>
<p>The autocorrelation captures correlations between residue properties at a given
distance over the whole sequence.</p>
<p>Based on the user-defined amino acid scale (see <a class="reference internal" href="feature_data_sources.html#feature-data-sources"><em>Feature data sources</em></a>),
a protein sequence is first translated to a property profile. Than, for a given
distance between two residues (the <em>lag</em> parameter), the method traverses over
the protein sequence, calculating a correlation measure for all residue pairs
at distance <em>lag</em>. Three different autocorrelation calculation types are
offered (the <em>type</em> parameter), all using a different correlation measure:
normalized Moreau-Broto <a class="reference internal" href="#moreau1980" id="id9">[12]</a>, Moran <a class="reference internal" href="#moran1950" id="id10">[11]</a>, and Geary
<a class="reference internal" href="#geary1954" id="id11">[6]</a>.</p>
<p>The implementation is based on the description is <a class="reference internal" href="#li2006" id="id12">[4]</a>, more
information about the implementation details can be found there.</p>
</div>
<div class="section" id="signal-average">
<h2>Signal average<a class="headerlink" href="#signal-average" title="Permalink to this headline">¶</a></h2>
<p>This feature translates a protein sequence into a property profile using
a user-defined amino acid scale (see <a class="reference internal" href="feature_data_sources.html#feature-data-sources"><em>Feature data sources</em></a>), and
calculates the average profile (signal) value as feature.</p>
<p>The <em>window</em> and <em>edge</em> parameter determine how much the profile is smoothed
before calculating the feature. A larger window results in more a more smoothed
profile. The edge determine how much influence the residues at the edge of the
window have on the smoothing (see also Fig.1B in <a class="reference internal" href="feature_data_sources.html#feature-data-sources"><em>Feature data sources</em></a>).</p>
<a class="reference internal image-reference" href="_images/featcalc3.png"><img alt="_images/featcalc3.png" class="align-center" src="_images/featcalc3.png" style="width: 640px;" /></a>
<p>Amino acid scales relate to different amino acid properties, such as
hydropathicity. The average value of such a scale therefore provides an
indication of the global hydropathicity of the protein.</p>
</div>
<div class="section" id="signal-peaks-area">
<h2>Signal peaks area<a class="headerlink" href="#signal-peaks-area" title="Permalink to this headline">¶</a></h2>
<p>The same as the previous feature, but instead of taking the average profile
value, this feature calculates the area under the profile curve under and above
some given threshold (see also Fig.1C in <a class="reference internal" href="feature_data_sources.html#feature-data-sources"><em>Feature data sources</em></a>).</p>
<a class="reference internal image-reference" href="_images/featcalc4.png"><img alt="_images/featcalc4.png" class="align-center" src="_images/featcalc4.png" style="width: 640px;" /></a>
</div>
<div class="section" id="pseudo-amino-acid-composition-type-1">
<h2>Pseudo amino acid composition type 1<a class="headerlink" href="#pseudo-amino-acid-composition-type-1" title="Permalink to this headline">¶</a></h2>
<p>The type 1 pseudo amino acid composition calculates 20 + lambda features as
introduced in <a class="reference internal" href="#chou2001" id="id13">[2]</a> and provides the same calculation as provided on
the PseAAC webserver <a class="reference internal" href="#shen2008" id="id14">[6]</a>. The same amino acid scales as on the
PseAAC webserver are also provided, of which one or more can be selected.</p>
<a class="reference internal image-reference" href="_images/feat_pseaac1.png"><img alt="_images/feat_pseaac1.png" class="align-center" src="_images/feat_pseaac1.png" style="width: 640px;" /></a>
</div>
<div class="section" id="pseudo-amino-acid-composition-type-2">
<h2>Pseudo amino acid composition type 2<a class="headerlink" href="#pseudo-amino-acid-composition-type-2" title="Permalink to this headline">¶</a></h2>
<p>The type 2 pseudo amino acid composition calculates 20 + lambda features as
introduced in <a class="reference internal" href="#chou2005" id="id15">[3]</a> and provides the same calculation as provided on
the PseAAC webserver <a class="reference internal" href="#shen2008" id="id16">[6]</a>. The same amino acid scales as on the
PseAAC webserver are also provided, of which one or more can be selected.</p>
<a class="reference internal image-reference" href="_images/feat_pseaac2.png"><img alt="_images/feat_pseaac2.png" class="align-center" src="_images/feat_pseaac2.png" style="width: 640px;" /></a>
</div>
<div class="section" id="quasi-sequence-order-descriptors">
<h2>Quasi sequence-order descriptors<a class="headerlink" href="#quasi-sequence-order-descriptors" title="Permalink to this headline">¶</a></h2>
<p>The quasi-sequence-order descriptors are introduced in <a class="reference internal" href="#chou2000" id="id17">[4]</a>.</p>
<a class="reference internal image-reference" href="_images/feat_qso.png"><img alt="_images/feat_qso.png" class="align-center" src="_images/feat_qso.png" style="width: 640px;" /></a>
</div>
<div class="section" id="secondary-structure-composition">
<h2>Secondary structure composition<a class="headerlink" href="#secondary-structure-composition" title="Permalink to this headline">¶</a></h2>
<p>Secondary structure sequences may contain three letters, the same that are used
by secondary structure prediction method PSIPRED: C (random coil), H (helix),
E (strand). This feature calculates the composition of these three letters.</p>
<a class="reference internal image-reference" href="_images/feat_ssc.png"><img alt="_images/feat_ssc.png" class="align-center" src="_images/feat_ssc.png" style="width: 640px;" /></a>
</div>
<div class="section" id="per-secondary-structure-amino-acid-composition">
<h2>Per secondary structure amino acid composition<a class="headerlink" href="#per-secondary-structure-amino-acid-composition" title="Permalink to this headline">¶</a></h2>
<p>This feature category calculates the amino acid composition per secondary
structure element.</p>
<a class="reference internal image-reference" href="_images/feat_ssaac.png"><img alt="_images/feat_ssaac.png" class="align-center" src="_images/feat_ssaac.png" style="width: 640px;" /></a>
<p>For example, for the following amino acid and corresponding secondary structure
sequence:</p>
<div class="highlight-python"><pre>protein:     A A A A A A F F F F F F D D D D D D
             | | . . . . . . * * * * . . | | | |
sec. struct: C C H H H H H H E E E E H H C C C C</pre>
</div>
<p>The sequences will be split per secondary structure type, gathering all
residues that are annotated with the same secondary structure letter:</p>
<div class="highlight-python"><pre>C (random coil): A A D D D D

H (helix):       A A A A F F D D

E (strand):      F F F F</pre>
</div>
<p>The composition of these three subsequence is than calculated and returned as
features:</p>
<div class="highlight-python"><pre>.    |        coil        |       helix        |       strand       |
        CA     CD     CF     HA     HD     HF     EA     ED     EF

       0.33   0.66   0.00   0.50   0.25   0.25   0.00   0.00   1.00</pre>
</div>
<p>This example only shows the composition of three amino acids (A,F,D), resulting
in 3 x 3 = 9 features. When using all amino acids, 3 x 20 = 60 features are
returned.</p>
</div>
<div class="section" id="solvent-accessibility-composition">
<h2>Solvent accessibility composition<a class="headerlink" href="#solvent-accessibility-composition" title="Permalink to this headline">¶</a></h2>
<p>Solvent accessibility sequences may contain two letters: B (buried),
E (exposed). This feature calculates the composition of these two letters.</p>
<a class="reference internal image-reference" href="_images/feat_sac.png"><img alt="_images/feat_sac.png" class="align-center" src="_images/feat_sac.png" style="width: 640px;" /></a>
</div>
<div class="section" id="per-solvent-accessibility-class-amino-acid-composition">
<h2>Per solvent accessibility class amino acid composition<a class="headerlink" href="#per-solvent-accessibility-class-amino-acid-composition" title="Permalink to this headline">¶</a></h2>
<p>This feature category calculates the amino acid composition per solvent
accessibility class (buried/exposed).</p>
<a class="reference internal image-reference" href="_images/feat_saaac.png"><img alt="_images/feat_saaac.png" class="align-center" src="_images/feat_saaac.png" style="width: 640px;" /></a>
<p>For example, for the following amino acid and corresponding solvent
accessibility sequence:</p>
<div class="highlight-python"><pre>protein:     A A A A A A F F F F F F D D D D D D
             . . . . | | | | . . | | | | . . . .
solvaccess:  E E E E B B B B E E B B B B E E E E</pre>
</div>
<p>The sequences will be split in a list with buried and a list with exposed amino
acids:</p>
<div class="highlight-python"><pre>E: A A A A F F D D D D

B: A A F F F F D D</pre>
</div>
<p>The composition of these two subsequence is than calculated and returned as
features:</p>
<div class="highlight-python"><pre>.    |      exposed       |       buried       |
        EA     ED     EF     BA     BD     BF

       0.40   0.40   0.20   0.25   0.25   0.50</pre>
</div>
<p>This example only shows the composition of three amino acids (A,F,D), resulting
in 2 x 3 = 6 features. When using all amino acids, 2 x 20 = 40 features are
returned.</p>
</div>
<div class="section" id="codon-composition">
<h2>Codon composition<a class="headerlink" href="#codon-composition" title="Permalink to this headline">¶</a></h2>
<p>This feature category calculates the relative frequency of occurrence of each
of the 64 codons in a protein&#8217;s ORF sequence.</p>
<a class="reference internal image-reference" href="_images/feat_cc.png"><img alt="_images/feat_cc.png" class="align-center" src="_images/feat_cc.png" style="width: 640px;" /></a>
</div>
<div class="section" id="codon-usage">
<h2>Codon usage<a class="headerlink" href="#codon-usage" title="Permalink to this headline">¶</a></h2>
<p>This feature category calculates the relative usage for each codon per amino
acid in the protein sequence.</p>
<a class="reference internal image-reference" href="_images/feat_cu.png"><img alt="_images/feat_cu.png" class="align-center" src="_images/feat_cu.png" style="width: 640px;" /></a>
<p>To illustrate this, consider the following protein amino acid sequence,
consisting of only alanines, and the
corresponding ORF sequence:</p>
<div class="highlight-python"><pre>amino acid:    A   A   A   A   A   A   A   A   A   A
       ORF:   GCC GCC GCC GCC GCC GCA GCA GCA GCT GCT</pre>
</div>
<p>Since four different codons encode for alanine: GCT, GCC, GCA, and GCG, the
example ORF sequence can only consist of these 4 codons. The codon usage
feature calculates the relative frequency of occurrence of these four codons,
resulting in the following 4 feature values:</p>
<div class="highlight-python"><pre>.       GCT    GCC    GCA    GCG
        0.2    0.5    0.3    0.0</pre>
</div>
<p>All other codon values will be set to 0.0 in this example. In a real sequence,
containing all 20 amino acids, this procedure is done for each amino acid,
resulting in a total of 64 features, one per codon.</p>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p id="bibtex-bibliography-features_spice-0"><table class="docutils citation" frame="void" id="cao2013" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[1]</a></td><td>D.S. Cao, Q.S. Xu, and Y.Z. Liang. Propy: a tool to generate various modes of chou’s PseAAC. <em>Bioinformatics</em>, 29(7):960–962, 2013.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="chou2001" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id13">[2]</a></td><td>K.C. Chou. Prediction of protein cellular attributes using pseudo-amino acid composition. <em>Proteins: Structure, Function, and Bioinformatics</em>, 43(3):246–255, 2001.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="chou2005" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id15">[3]</a></td><td>K.C. Chou. Using amphiphilic pseudo amino acid composition to predict enzyme subfamily classes. <em>Bioinformatics</em>, 21(1):10–19, 2005.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="chou2000" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id17">[4]</a></td><td>Kuo-Chen Chou. Prediction of protein subcellular locations by incorporating quasi-sequence-order effect. <em>Biochemical and Biophysical Research Communications</em>, 278(2):477–483, 2000.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="dubchak1995" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id7">[5]</a></td><td>I.&nbsp;Dubchak, I.&nbsp;Muchnik, S.R. Holbrook, and S.H. Kim. Prediction of protein folding class using global description of amino acid sequence. <em>Proceedings of the National Academy of Sciences</em>, 92(19):8700–8704, 1995.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="geary1954" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id11">[6]</a></td><td>R.C. Geary. The contiguity ratio and statistical mapping. <em>The Incorporated Statistician</em>, 5(3):115–146, 1954.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="georgiev2009" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[1]</td><td>A.G. Georgiev. Interpretable numerical descriptors of amino acid space. <em>Journal of Computational Biology</em>, 16(5):703–723, 2009.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="kawashima2008" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[2]</td><td>S.&nbsp;Kawashima, P.&nbsp;Pokarowski, M.&nbsp;Pokarowska, A.&nbsp;Kolinski, T.&nbsp;Katayama, and M.&nbsp;Kanehisa. AAindex: amino acid index database, progress report 2008. <em>Nucleic Acids Research</em>, 36(suppl 1):D202–D205, 2008.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="kyte1982" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[3]</td><td>J.&nbsp;Kyte and Russell&nbsp;F Doolittle. A simple method for displaying the hydropathic character of a protein. <em>Journal of Molecular Biology</em>, 157(1):105–132, 1982.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="li2006" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[4]</td><td><em>(<a class="fn-backref" href="#id3">1</a>, <a class="fn-backref" href="#id8">2</a>, <a class="fn-backref" href="#id12">3</a>)</em> Ze-Rong Li, Hong&nbsp;Huang Lin, LY&nbsp;Han, L&nbsp;Jiang, X&nbsp;Chen, and Yu&nbsp;Zong Chen. PROFEAT: a web server for computing structural and physicochemical features of proteins and peptides from amino acid sequence. <em>Nucleic Acids Research</em>, 34(suppl 2):W32–W37, 2006.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="moran1950" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id10">[11]</a></td><td>P.A.P. Moran. Notes on continuous stochastic phenomena. <em>Biometrika</em>, 37(1/2):17–23, 1950.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="moreau1980" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id9">[12]</a></td><td>G.&nbsp;Moreau and P&nbsp;Broto. Autocorrelation of molecular structures, application to SAR studies. <em>Nouveau Journal de Chimie</em>, 4:757–764, 1980.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="rao2011" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[13]</a></td><td>H.B. Rao, F.&nbsp;Zhu, G.B. Yang, Z.R. Li, and Y.Z. Chen. Update of PROFEAT: a web server for computing structural and physicochemical features of proteins and peptides from amino acid sequence. <em>Nucleic Acids Research</em>, 39(suppl 2):W385–W390, 2011.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="schneider1994" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[5]</td><td>G.&nbsp;Schneider and P.&nbsp;Wrede. The rational design of amino acid sequences by artificial neural networks and simulated molecular evolution: de novo design of an idealized leader peptidase cleavage site. <em>Biophysical Journal</em>, 66(2):335–344, 1994.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="shen2008" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[6]</td><td><em>(<a class="fn-backref" href="#id5">1</a>, <a class="fn-backref" href="#id14">2</a>, <a class="fn-backref" href="#id16">3</a>)</em> H.B. Shen and K.C. Chou. PseAAC: a flexible web server for generating various kinds of protein pseudo amino acid composition. <em>Analytical Biochemistry</em>, 373(2):386–388, 2008.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="vandenberg2010" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[16]</a></td><td>B.&nbsp;A. van&nbsp;den Berg, J.&nbsp;F. Nijkamp, M.&nbsp;J.&nbsp;T. Reinders, L.&nbsp;Wu, H.&nbsp;J. Pel, J.&nbsp;A. Roubos, and D.&nbsp;De&nbsp;Ridder. Sequence-based prediction of protein secretion success in \textit Aspergillus niger. In <em>Proceedings of Pattern Recegnition in Bioinformatics 2010</em>, 3–14. Springer, 2010.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="vandenberg2012" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[17]</a></td><td>B.A. van&nbsp;den Berg, M.J.T. Reinders, M.&nbsp;Hulsman, L.&nbsp;Wu, H.J. Pel, J.A. Roubos, and D.&nbsp;de&nbsp;Ridder. Exploring sequence characteristics related to high-level production of secreted proteins in \textit Aspergillus niger. <em>PLoS ONE</em>, 7(10):e45869, 2012.</td></tr>
</tbody>
</table>
</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="file_formats.html" title="File Formats"
             >next</a> |</li>
        <li class="right" >
          <a href="feature_data_sources.html" title="Feature data sources"
             >previous</a> |</li>
    
        <li><a href="index.html">SPiCE 0.1 documentation</a> &raquo;</li>
 
      </ul>
    </div>
    
    <div class="footer">
        &copy; Copyright 2013, Bastiaan van den Berg.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b1.
    </div>
    <div class="spica_footer">
    Delft University of Technology, <a href="http://bioinformatics.tudelft.nl">Delft Bioinformatics Lab</a>
    </div>

  </body>
</html>